pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -108,19 +112,19 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 		// The object is not being deleted, so if it does not have a finalizer,
 		// then lets add the finalizer and update the object.
 		if !helper.ContainsString(instance.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName) {
-			err = r.addFinalizerAndUpdate(&request)
+			err = r.addFinalizerAndUpdate(&request, logger)
```

Comment:
I am tented to sugget to have requestLogger as member of `PolicyReconcile` since we have only one goroutine running `Reconcile` and finalizer functions, but that's wrong if we do so and we change that we are broken, the alternative so we don't pass the logger to functions would be to add a struct with `PolicyReconcile`and logger and addFinalizer and removeFinalizers are functions from that struct.

Reply:
The point of this PR is to add unique ID per each execution of Reconcile, so we can track those.